CRM/Contribute/BAO/Contribution.php

```diff
@@ -4545,12 +4545,7 @@ public static function completeOrder($input, &$ids, $objects, $isPostPaymentCrea
       CRM_Contribute_BAO_ContributionRecur::addrecurSoftCredit($objects['contributionRecur']->id, $contribution->id);
     }
 
-    if (empty($contribution->_relatedObjects['participant']) && !empty($contribution->_relatedObjects['membership'])) {
-      // @fixme Can we remove this if altogether? - we removed the participant if / else and left relatedObjects['participant'] to ensure behaviour didn't change but it is probably not required.
-      // @todo - use getRelatedMemberships instead
-      $contribution->trxn_id = $input['trxn_id'] ?? NULL;
-      $contribution->receive_date = CRM_Utils_Date::isoToMysql($contribution->receive_date);
-    }
+    $contribution->trxn_id = $input['trxn_id'] ?? NULL;
```

Comment:
@eileenmcnaughton This line can go too - we don't pass $contribution by reference anymore and the only place it's passed is later in this function to `CRM_Activity_BAO_Activity::addActivity` which doesn't use trxn_id.

Reply:
yep agree - I've removed it. CRM_Activity_BAO_Activity::addActivity requires a lot of our remaining vars